### ISSUE TITLE
Vault API endpoint /v1/auth/cert/login returns HTTP 500 Internal Server Error on failover

### ISSUE THREAD

**Issue by ryan-talley** [hashicorp/vault#6595]
**Describe the bug**
This is occurring on a 3 node cluster with Vault and etcd running on each node. etcd is configured to run in HA, however, the vault servers do not have HA enabled. Each vault server is configured to talk with the etcd instance on its node only on the localhost. The vault API endpoint /v1/auth/cert/login returns HTTP 500 Internal Server Error during a failover. This endpoint is reachable (HTTP 200 OK) before the failover occurs and awhile (~20min) after the failover. Other API endpoints such as /v1/sys/health are available during the failover. etcd endpoints such as /health are also reachable (HTTP 200 OK). Requests to this endpoint will hang for ~5 seconds before returning the 500 error. Vault log contains ERRORs with context deadline exceeded and is attached below.

**To Reproduce**
1. Vault is running on all 3 cluster nodes with the provided config along with etcd
2. Run `curl --key /etc/vault/ssl/vault-<redacted>-key.pem --cert /etc/vault/ssl/vault-<redacted>.pem --cacert /etc/vault/ssl/ca.pem -X PUT https://<redacted>:8200/v1/auth/cert/login` on the failover node
3. See it return HTTP 200 OK
4. Power off the current master node
5. Repeat step 2
6. See it return HTTP 500 Internal Server Error after hanging for ~5sec
7. Run `curl --key /etc/vault/ssl/vault-<redacted>-key.pem --cert /etc/vault/ssl/vault-<redacted>.pem --cacert /etc/vault/ssl/ca.pem https://<redacted>:8200/v1/sys/health` on the failover node
8. See it return HTTP 200 OK
9. Run `curl --key /etc/etcd/ssl/node-<redacted>key.pem --cert /etc/etcd/ssl/node-<redacted>.pem --cacert /etc/etcd/ssl/ca.pem https://127.0.0.1:2379/health` on the failover node
10. See it return HTTP 200 OK
11. Wait ~20min
12. Repeat step 2
13. See it return HTTP 200 OK

**Expected behavior**
This API endpoint should not hang and return 500 errors during a failover allowing services to authenticate with vault.

**Environment:**
* Vault Server Version (retrieve with `vault status`): Vault v0.11.4
* Vault CLI Version (retrieve with `vault version`): N/A (CLI not used)
* Server Operating System/Architecture:
Docker container: docker.io/vault:0.11.4 ae66482e5cf4
`uname -a`: Linux <redacted> 3.10.0-862.11.6.el7.x86_64
CentOS Atomic 7.1808

Vault server configuration file(s):
These files are identical across the 3 nodes except for their IPs which have been redacted
```hcl
storage "etcd" {
  address = "https://127.0.0.1:2379"
  path = "vault"
  tls_cert_file = "/etc/etcd/ssl/node-<redacted>.pem"
  tls_key_file = "/etc/etcd/ssl/node-<redacted>key.pem"
  tls_ca_file = "/etc/etcd/ssl/ca.pem"
}

listener "tcp" {
  address = "0.0.0.0:8200"
  cluster_address = "<redacted>:8201"
  tls_disable = 0
  tls_cert_file = "/etc/vault/ssl/vault-<redacted>.pem"
  tls_key_file = "/etc/vault/ssl/vault-<redacted>-key.pem"
}

cluster_addr = "https://<redacted>:8201"
default_lease_ttl = "2160h"
max_lease_ttl = "8760h"
```

**Additional context**
Attached is the redacted vault log from the node that the requests were being made from.
[vault.log](https://github.com/hashicorp/vault/files/3087124/vault.log)

**Comment by ryan-talley** [comment]
I am closing this issue, since this turned out to be an issue with our etcd cluster backend.

TL;DR: running etcd clusters that broadcast to their peers on the same subnet caused this issue. This was resolved by moving those clusters to their own isolated subnets.

If anyone is interested in what happened we were running several of these vault clusters each with their own backing etcd cluster. The etcd clusters were configured to broadcast to peers on their IP on port 2380. All of these etcd clusters were part of the same subnet. Running tcpdump on nodes in these clusters revealed that traffic from other etcd clusters were ending up on the other etcd clusters. This high volume of traffic was slowing down the etcd responses to vault causing it to lock up when hitting the endpoint I described. The etcd clusters were not sharing information, however, since they were using cluster tokens so they could identify their cluster peers. When the etcd clusters were isolated on their own subnets this issue went away and vault worked as expected.
